### ISSUE TITLE
Decide the ideal configuration for the vehicle

### ISSUE THREAD

**Issue by Diapolo10** [Diapolo10/rl_car#9]
The config file has a lot of parameters that could be adjusted, the current ones are more like placeholders. Someone should play around with the values until a combination is found that would enable "natural"-looking driving while not making the track impossible.

This should be done before the neural network training process begins.

**Comment by kearkyle** [comment]
Hi @Diapolo10, can I get assigned for this issue ?

**Comment by Diapolo10** [comment]
> Hi @Diapolo10, can I get assigned for this issue ?

So, here's the thing; this is kind of supposed to be a team project for one of my university courses - I just made it public so that I don't have to separately invite everyone as I saw no value in keeping the project private anyway. I don't _mind_ if you want to help, but the majority of the work will likely be done by my team members.

I'll assign you because I see no harm in doing it, but you choose for yourself if you want to help or not.

**Comment by kearkyle** [comment]
Thanks for letting me know, since it's a school project I don't want to affect your grades or anything, I'll just work on this issue if that's ok with you or if you guys are working on this issue yourselves then please let me know.

**Comment by Diapolo10** [comment]
> Thanks for letting me know, since it's a school project I don't want to affect your grades or anything, I'll just work on this issue if that's ok with you or if you guys are working on this issue yourselves then please let me know.

Honestly I don't think our grades are really an issue, the one in charge is pretty lenient, so don't worry about that. You could say that my main concern is that the team is fairly small (two people right now, we may get a third) and I would like to see everyone have _something_ to do. Personally I've been doing most of the heavy lifting.

**Comment by kearkyle** [comment]
So I did multiple attempts and this is what I changed to make it move more natural. I only changed the configs under `#Game`, the bottom configs are intended for the AI is that correct? When I open the game there's only 1 red car, so am I missing anything? Is there like a button to start the game or something or this is all we have so far?
Here's the changes I did: https://github.com/kearkyle/rl_car/commit/19a617194b90bf0fc7555c938734523221441649

**Comment by Diapolo10** [comment]
> _the bottom configs are intended for the AI is that correct?_

Yes, that's correct. Well, more accurately they're placeholders from another project and may end up going even entirely unused, but that's my problem. :p

> _When I open the game there's only 1 red car, so am I missing anything?_

No, I've focused on debugging the vision system so I haven't put forth much effort to set the simulation up for multiple agents yet. Though [there is an issue set up for that feature](https://github.com/Diapolo10/rl_car/issues/10).

> _Is there like a button to start the game or something or this is all we have so far?_

I saw no value in adding anything like that, because this isn't something intended to be played by a human. The "game" exists as a simulation for the neural network to learn how to navigate it while accumulating rewards, with the end goal of having it try and race across the track as quickly as it can.

This project is fairly simple in that I don't currently care about overfitting the neural net, as the scope isn't large enough to implement multiple randomised tracks at this point. That could be a future feature if we decide to continue this on some other course, or on our free time, but right now the main goal is to just have something that works.

> _Here's the changes I did: https://github.com/kearkyle/rl_car/commit/19a617194b90bf0fc7555c938734523221441649_

I'll look at those changes in more detail this weekend, though at first glance I'm a bit sceptical about changing the sprite scaling right now.

**Comment by kearkyle** [comment]
My bad with the sprite scaling, if you want I can do tests on the original size or adjust it to your preferences. Also for the multiplayer issue is your team working on it? I can give it a try if you want.

**Comment by Diapolo10** [comment]
> _My bad with the sprite scaling, if you want I can do tests on the original size or adjust it to your preferences._

It's fine, I approved it and already made and merged a pull request from your fork. Thanks btw.

> Also for the multiplayer issue is your team working on it? I can give it a try if you want.

I just assigned my partner to work on a small part of it (the issue I linked above), but I came to realise that actually letting multiple agents to control vehicles means an extra step is needed; something to bypass the keyboard key requirement and instead pass "button presses" directly. Since I just thought of that I haven't even written an additional issue yet, but that one is probably going to be more difficult.

If you want, you can go to the other thread and discuss the current issue with my partner. We don't have any set schedules for working on this, only a deadline.

That said I don't think "multiplayer" is the right word for it considering this is basically going to be a neural net boot camp and none of the individual agents can affect each other. Plus it's kind of just a "nice to have" feature that would speed up teaching the network.

**Comment by Diapolo10** [comment]
Oh, and I didn't close this issue yet because I'm not sure if this is the best configuration yet. Though I do think it beats the previous one.
